src/components/BarChart.js

```diff
@@ -0,0 +1,56 @@
+import { extent, max } from 'd3-array'
+import { scaleLinear, scaleTime } from 'd3-scale'
+import { timeParse } from 'd3-time-format'
+import React from 'react'
+
+import XAxis from './XAxis'
+import YAxis from './YAxis'
+
+const BarChart = ({
+  size = { width: 600, height: 400 },
+  margin = { top: 20, right: 20, bottom: 30, left: 50 },
+  labels = true,
+  data,
+}) => {
+  const height = size.height - margin.top - margin.bottom
+  const parse = timeParse('%Y-%m-%d')
+  const dataClean = data.map(d => ({ date: parse(d[0]), value: +d[1] }))
+  const width = size.width - margin.left - margin.right
+
+  const x = scaleTime()
```

Comment:
for bar charts, i think we want the x scale to be ordinal (https://github.com/d3/d3-scale#band-scales), at least by default -- this will fix the bar over the axis issue

perhaps something like this:
```
import { scaleBand } from 'd3-scale'

const x = scaleBand()
    .domain(dataClean.map(d => d.date))
    .rangeRound([0, width]).padding(0.1)
```

for ref: http://bl.ocks.org/mbostock/3885304, http://bl.ocks.org/mbostock/5977197